RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,301 @@
+A CN that is also a BN should maintain more peer connections with other BNs, but should also have some connections with other VNs.
+
+Having some connections with VNs are important as BNs have derived node ids and not registered node ids such as VNs, making it possible for the CN to be separated from the main network and become victim to an eclipse attack.
+Having some connections with VNs will make it more difficult to separate the CN from the network and will ensure successful propagation of transactions and completed blocks from that CN. 
+
+A CN that is also a VN should maintain more peer connections with other VNs, but also have some connections with BNs.
+CNs that are part of Validator Node committees should attempt to maintain permanent connections with the other members of the committee to ensure that quick consensus can be achieved.
+
+To maintain connections with peers, the following process can be performed.
+Once some peers have been found using discovery requests, resulting in them successfully being added to the local routing table.
+The CN can decide how the peer connections should be selected from the routing table by either:
+ - manually selecting a subset,
+ - automatically selecting a random subset or
+ - selecting a subset of neighbouring nodes with similar node ids. 
+
+ Maintaining communication channels are important and the following process can be followed in an attempt to keep peer connections alive:
+For an existing peer connection.
+When more than 30 minutes have passed since the last communication with that peer, then a heartbeat message should be sent to the peer in an attempt to keep the connection with the peer alive.
+If that specific peer connection is not important then a new peer can be selected from the local routing table.
+A new connection can then be established and maintained between the current node and the newly selected node.
+If that specific connection is important, such as with the connections between committee members, then the current CN or CC must wait and attempt to create a new connection with that same peer.
+If more than 90 minutes have passed since the last successful communication with the peer node, a new discovery request can be sent on the Tari communication network in an attempt to locate that peer again.
+Losing of a peer might happen in cases where the CN or CC went temporarily offline and their dynamic communication address changed, requiring the discovery process to be performed again before a direct P2P communication channel can be established.
+
+#### Functionality Required of Consensus Nodes
+
+- It MUST select a cryptographic key pair used for identification on the Tari Communication network.
+- A CN MAY request the peer addresses of CNs with similar node ids from other CNs to extend their local routing table. 
+- If a CN is a VN, then a node id MUST be obtained by registering on the Base layer.
```

Comment:
what happens if there is a clash?

Reply:
I might be incorrect, but I think the likelihood of a clash is the same as the likelihood of generating the same private key as another user. If a VN discovers a clash directly after registration, then he might have to reregister to obtain a new node id. I think registered node ids should have priority over derived node ids. Do you have any suggestions how we could deal with this?

Reply:
We may need to specify hashing of some sort of entropy to be included in the cryptographic key pair.

Reply:
If a node ID is some hash(public key || ...) there's an infinitesimal chance of a clash. The simplest algo would be
```
  Generate id = H(PK || ...)
  If clash (which happens every 1 in 2^200 times or something), try again with:  a new set of keys, or modify "Other data"
```